### PR TITLE
Add validations to ImmunisationImport

### PR DESCRIPTION
**Pull request by tvararu**
This adds a few of the basic validations that we'll need for the NIVS upload journey.

### CODE REVIEW

File: app/models/immunisation_import.rb [hunk 1]
```diff
@@ -21,16 +21,32 @@
...
+    return if invalid?
+
+    self.data ||= CSV.parse(csv, headers: true, skip_blanks: true)
+  rescue CSV::MalformedCSVError
+    self.csv_is_malformed = true
+  ensure
+    csv.close if csv.respond_to?(:close)
+  end
+
   def process!(patient_session:)
```
Comment (thomasleese) [inline app/models/immunisation_import.rb:49]: Should this `raise` or something similar if we haven't previously called `load_data!`, i.e. `data` is `nil`?

File: spec/models/immunisation_import_spec.rb [hunk 2]
```diff
@@ -30,8 +30,45 @@
 
   it { should validate_presence_of(:csv) }
 
+  describe "#load_data!" do
+    describe "with malformed CSV" do
+      let(:file) { "malformed.csv" }
+
+      it "is invalid" do
+        subject.load_data!
```
Comment (thomasleese) [inline spec/models/immunisation_import_spec.rb:38]: This could go in a `before` block since every test does it.

**Review by thomasleese — APPROVED** [review]
Looks good to me, just a couple of comments, but not blocking.

File: app/models/immunisation_import.rb (hunk 1 above)
Comment (tvararu) [reply app/models/immunisation_import.rb:49]: What about calling `load_data!` if `data.nil?`?

Comment (thomasleese) [reply app/models/immunisation_import.rb:49]: Yeah that would work too, although it would still then need to check the data was loaded correctly.

**Pull request merged**